client/src/connection_cache.rs

```diff
@@ -60,54 +50,33 @@ pub fn set_use_quic(use_quic: bool) {
     map.set_use_quic(use_quic);
 }
 
-#[allow(dead_code)]
 // TODO: see https://github.com/solana-labs/solana/issues/23661
 // remove lazy_static and optimize and refactor this
 fn get_connection(addr: &SocketAddr) -> Connection {
     let mut map = (*CONNECTION_MAP).lock().unwrap();
-    let ticks = map.ticks;
-    let use_quic = map.use_quic;
-    let (conn, target_ticks) = match map.map.entry(*addr) {
-        Entry::Occupied(mut entry) => {
-            let mut pair = entry.get_mut();
-            let old_ticks = pair.1;
-            pair.1 = ticks;
-            (pair.0.clone(), old_ticks)
+
+    match map.map.get(addr) {
+        Some(connection) => {
+            inc_new_counter_info!("get_connection-cache-hits", 1);
+            connection.clone()
         }
-        Entry::Vacant(entry) => {
+        None => {
+            inc_new_counter_info!("get_connection-cache-misses", 1);
             let (_, send_socket) = solana_net_utils::bind_in_range(
                 IpAddr::V4(Ipv4Addr::new(0, 0, 0, 0)),
                 VALIDATOR_PORT_RANGE,
             )
             .unwrap();
-            let conn = if use_quic {
+            let connection = if map.use_quic {
                 Connection::Quic(Arc::new(QuicTpuConnection::new(send_socket, *addr)))
             } else {
                 Connection::Udp(Arc::new(UdpTpuConnection::new(send_socket, *addr)))
             };
 
-            entry.insert((conn.clone(), ticks));
-            (conn, ticks)
+            map.map.put(addr.clone(), connection.clone());
```

Comment:
```suggestion
            map.map.put(*addr, connection.clone());
```

`SockAddr` is `Copy`. Odd that clippy didn't complain :thinking: